src/rust/aggregator/src/pipelines/coins.rs

```diff
@@ -0,0 +1,138 @@
+            r#"
+                WITH market_coins AS (
+                    SELECT DISTINCT
+                        base_account_address AS address,
+                        base_module_name AS module,
+                        base_struct_name AS struct
+                    FROM
+                        market_registration_events
+                    WHERE base_account_address IS NOT NULL
+                    AND base_module_name IS NOT NULL
+                    AND base_struct_name IS NOT NULL
+                    UNION
+                    SELECT DISTINCT
+                        quote_account_address AS address,
+                        quote_module_name AS module,
+                        quote_struct_name AS struct
+                    FROM
+                        market_registration_events
+                )
+                SELECT
+                    *
+                FROM
+                    market_coins AS m
+                WHERE
+                    NOT EXISTS (
+                        SELECT
+                            *
+                        FROM
+                            aggregator.coins AS c
+                        WHERE
+                            c.address = m.address
+                        AND
+                            c.module = m.module
+                        AND
+                            c.struct = m.struct
+                    )
+            "#,
+        )
+        .fetch_all(&self.pool)
+        .await
+        .map_err(|e| PipelineError::ProcessingError(anyhow!(e)))?;
+        for coin in coins {
+            let (address, module, struct_name) = (coin.address.unwrap(), coin.module.unwrap(), coin.r#struct.unwrap());
+            let move_struct_tag_str = format!("{}::{}::{}", address, module, struct_name);
+            let coin_info = get_coin_info(move_struct_tag_str).await;
+            match coin_info {
+                Ok(coin_info) => {
+                    let CoinInfo { name, symbol, decimals } = coin_info;
+                    sqlx::query!(
+                        "INSERT INTO aggregator.coins VALUES ($1, $2, $3, $4, $5, $6)",
+                        name, symbol, decimals, address, module, struct_name
+                    )
+                    .execute(&self.pool)
+                    .await
+                    .map_err(|e| PipelineError::ProcessingError(anyhow!(e)))?;
+                },
+                Err(GetCoinInfoError::RateLimited) => {},
+                Err(GetCoinInfoError::Other(e)) => {return Err(PipelineError::ProcessingError(anyhow!(e)));},
+            }
+        }
+        Ok(())
+    }
+}
+
+struct CoinInfo {
+    name: String,
+    symbol: String,
+    decimals: i16,
+}
+
+#[derive(Debug, Error)]
+enum GetCoinInfoError {
+    #[error("Got rate limited by Aptos")]
+    RateLimited,
+    #[error("Other error: {0}")]
+    Other(anyhow::Error),
+}
+
+async fn get_coin_info(move_struct_tag_str: String) -> Result<CoinInfo, GetCoinInfoError> {
```

Comment:
To do this, query for the `0x1::coin::CoinInfo<CoinType>` under the coin address, then extract out the `name`, `symbol`, and `decimals` fields